services/QuillLMS/client/app/bundles/Staff/components/activityForm/topicGroup.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react'
+
+import TopicColumn from './topicColumn'
+
+const TopicGroup = ({ createNewTopic, findParentTopic, topic, topicOptions, handleLevelOneChange }) => {
+
+  const [levelOneTopic, setLevelOneTopic] = React.useState(topic)
+  const [levelTwoTopic, setLevelTwoTopic] = levelOneTopic.id ? React.useState(findParentTopic(topic)) : React.useState({})
+  const [levelThreeTopic, setLevelThreeTopic] = levelOneTopic.id ? React.useState(findParentTopic(levelTwoTopic)) : React.useState({})
+
+  React.useEffect(() => {
+    if (levelOneTopic.parent_id !== levelTwoTopic.id) {
+      handleLevelOneChange(levelOneTopic.id, null)
+      setLevelOneTopic({})
+    }
+  }, [levelTwoTopic]);
+
+  React.useEffect(() => {
+    if (levelTwoTopic.parent_id !== levelThreeTopic.id) {
+      setLevelTwoTopic({})
+    }
+  }, [levelThreeTopic]);
+
+  function selectTopic(topicId, level) {
+    const newTopic = topicOptions.find(t => t.id === topicId)
+    if (level === 1) {
+      handleLevelOneChange(levelOneTopic.id, newTopic.id)
+      setLevelOneTopic(newTopic)
+    } else if (level === 2) {
+      setLevelTwoTopic(newTopic)
+    } else if (level === 3) {
+      setLevelThreeTopic(newTopic)
+    }
```

Comment:
Since this `if` block basically gets repeated in the `removeTopic` function below, I wonder if it's worth setting up a const like:

```
const levelNumberToFunction = {
  1: setLevelOneTopic,
  2: setLevelTwoTopic,
  3: setLevelThreeTopic
}
```

And then here you could just call `levelNumberToTopic[level](newTopic)` and below you would call `levelNumberToTopic[level]({})`.